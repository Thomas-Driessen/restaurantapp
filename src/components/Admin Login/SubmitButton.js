import React from 'react';
import { Button } from '@material-ui/core';

class SubmitButton extends React.Component{

    render(){
        return(
            <div className = "submitButton">
                <Button
                    className ='btn'
                    disabled = {this.props.disabled}
                    onClick={() => this.props.onClick()}
                    variant="outlined"
                >
                    {this.props.text}
                </Button>
            </div>
        )
    }
}

export default SubmitButton;