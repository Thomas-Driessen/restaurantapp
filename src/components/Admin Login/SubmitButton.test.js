import React from 'react';
import { render } from '@testing-library/react';
import SubmitButton from './SubmitButton';
import renderer from 'react-test-renderer';

test('Render SubmitButton Restaurant', () => {
    const { getByText } = render(<SubmitButton text='Login'/>);
    const linkElement = getByText(/Login/i);
    expect(linkElement).toBeInTheDocument();
});

it('SubmitButton renders correctly', () => {
    const tree = renderer
        .create(<SubmitButton text='Login'/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});