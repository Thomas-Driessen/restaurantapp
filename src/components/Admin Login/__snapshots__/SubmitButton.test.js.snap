// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton renders correctly 1`] = `
<div
  className="submitButton"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined btn"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Login
    </span>
  </button>
</div>
`;
